### PR TITLE
MacOS Build

### PR DESCRIPTION
**Pull request by 007harshmahajan**
adding of `tor` libs for macos build and cleaning README.md

### CODE REVIEW

**Review by Shourya742 — CHANGES_REQUESTED** [review]
ACK. If possible, could you also address the CI/CD pipeline change related to #200

**Comment by mojoX911** [comment]
There is another way to include build scripts in the crate, with more granular control for platform-specific compilation. Maybe that could be easier, and simpler for users of the lib, than features?

https://doc.rust-lang.org/cargo/reference/build-scripts.html

File: Cargo.toml [hunk 1]
```diff
@@ -28,9 +28,11 @@ openssl-sys = { version = "0.9.68", optional = true }
 [features]
 default = ['tor']
 # Mac OS needs to use the vendored version of openssl-sys as upstream `libtor` is not updated for to support mac os aarch (M series)
-build-macos-aarch = ["openssl-sys/vendored"]
+build-macos-aarch = ['openssl-sys/vendored', 'dep:libtor', 'dep:mitosis']
 # The following feature set is in response to the issue described at https://github.com/rust-lang/rust/issues/45599
 # Only used for running the integration tests
 integration-test = []
+# Mac OS needs to use the vendored version of openssl-sys for integrating testing as upstream `libtor` is not updated for to support mac os aarch (M series)
+integration-test-macos-aarch = ['openssl-sys/vendored']
```
Comment (mojoX911) [inline Cargo.toml:36]: This also needs to be mentioned in the readme, as a note for the test command.

File: README.md [hunk 2]
```diff
@@ -49,7 +49,7 @@ CoinSwap is a rust implementation of a variant of atomic-swap protocol, using HT
 Macos (M series) needs some feature for compilation 
 
 ```console
-$ cargo build --target=aarch64-apple-darwin  --features=build-macos-aarch
+$ cargo build --features=build-macos-aarch
```
Comment (mojoX911) [inline README.md:52]: I don't know when we removed it, but we need to add the regular build command too in the readme.

The readme should have:
 - Command to build in regular Linux.
 - Note: Command for building on Mac.
 - Command to test in regular Linux.
 - Note: Command for testing on Mac.

**Review by mojoX911 — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 3]
```diff
@@ -28,9 +28,11 @@ openssl-sys = { version = "0.9.68", optional = true }
 [features]
 default = ['tor']
 # Mac OS needs to use the vendored version of openssl-sys as upstream `libtor` is not updated for to support mac os aarch (M series)
-build-macos-aarch = ["openssl-sys/vendored"]
+build-macos-aarch = ['openssl-sys/vendored', 'dep:libtor', 'dep:mitosis']
```
Comment (mojoX911) [inline Cargo.toml:31]: Why not?
```suggestion
build-macos-aarch = ['default', 'openssl-sys/vendored']
```

File: Cargo.toml (hunk 1 above)
Comment (mojoX911) [reply Cargo.toml:36]: Do we need another feature for this? 

We should be able to do just `cargo test --features=integration-test, build-macos-arch -- --no-cpature`?

File: Cargo.toml [hunk 4]
```diff
@@ -28,9 +28,11 @@ openssl-sys = { version = "0.9.68", optional = true }
...
 # The following feature set is in response to the issue described at https://github.com/rust-lang/rust/issues/45599
 # Only used for running the integration tests
 integration-test = []
+# Mac OS needs to use the vendored version of openssl-sys for integrating testing as upstream `libtor` is not updated for to support mac os aarch (M series)
+integration-test-macos-aarch = ['openssl-sys/vendored']
 # Only used for running the tor
 tor = ['dep:libtor', 'dep:mitosis']
```
Comment (mojoX911) [inline Cargo.toml:38]: This works too.  

```suggestion
tor = ['libtor', 'mitosis']
```

File: Cargo.toml [hunk 5]
```diff
@@ -28,9 +28,11 @@ openssl-sys = { version = "0.9.68", optional = true }
...
 default = ['tor']
 # Mac OS needs to use the vendored version of openssl-sys as upstream `libtor` is not updated for to support mac os aarch (M series)
-build-macos-aarch = ["openssl-sys/vendored"]
+build-macos-aarch = ['openssl-sys/vendored', 'dep:libtor', 'dep:mitosis']
 # The following feature set is in response to the issue described at https://github.com/rust-lang/rust/issues/45599
 # Only used for running the integration tests
 integration-test = []
+# Mac OS needs to use the vendored version of openssl-sys for integrating testing as upstream `libtor` is not updated for to support mac os aarch (M series)
+integration-test-macos-aarch = ['openssl-sys/vendored']
 # Only used for running the tor
```
Comment (mojoX911) [inline Cargo.toml:37]: Better framing
```suggestion
# Used for spawning Tor process and connecting to the Tor socks port`
```

We can later divide it into two separate features. Spawning and Connecting.

Not all users will need both. For some use cases, we may need only to connect tor socks port (mobile taker apps with Orbot). So mitosis can be avoided for those builds.

**Comment by mojoX911** [comment]
> ACK. If possible, could you also address the CI/CD pipeline change related to https://github.com/citadel-tech/coinswap/pull/200

Will be good to get it included in this PR itself.

File: Cargo.toml (hunk 1 above)
Comment (007harshmahajan) [reply Cargo.toml:36]: - Will update the docs
- testing with `--features=integration-test` doesnt include `tor` libs but do include the `openssl` which is causing issues for `integration-test` on macos, hence we need to have `macos specific integration testing` due to `openssl upstream`. Macos only with `aarch`

File: README.md (hunk 2 above)
Comment (007harshmahajan) [reply README.md:52]: -  Will update the docs

File: Cargo.toml (hunk 3 above)
Comment (007harshmahajan) [reply Cargo.toml:31]: ACK, this works, will update

File: Cargo.toml (hunk 5 above)
Comment (007harshmahajan) [reply Cargo.toml:37]: Looks great, will update

**Comment by 007harshmahajan** [comment]
failing integration test when running on macos with `cargo test --feature=integration-test-macos-aarch`
```test src/lib.rs - (line 77) ... FAILED

failures:

---- src/lib.rs - (line 77) stdout ----
error: unknown start of token: `
  --> src/lib.rs:85:47
   |
11 | Play through a single test case, for example, `standard_swap`,  by running:
   |                                               ^
   |
help: Unicode character '`' (Grave Accent) looks like ''' (Single Quote), but it is not
   |
11 | Play through a single test case, for example, 'standard_swap`,  by running:
   |                                               ~

error: unknown start of token: `
  --> src/lib.rs:85:61
   |
11 | Play through a single test case, for example, `standard_swap`,  by running:
   |                                                             ^
   |
help: Unicode character '`' (Grave Accent) looks like ''' (Single Quote), but it is not
   |
11 | Play through a single test case, for example, `standard_swap',  by running:
   |                                                             ~

error: unknown start of token: `
  --> src/lib.rs:87:1
   |
13 | ```console
   | ^^^
   |
   = note: character appears 2 more times
help: Unicode character '`' (Grave Accent) looks like ''' (Single Quote), but it is not
   |
13 | '''console
   | ~~~

error: expected one of `!`, `.`, `::`, `;`, `?`, `{`, `}`, or an operator, found `test`
 --> src/lib.rs:80:6
  |
6 | Each test in the [tests](./tests/) folder covers a different edge-case situation and demonstrates how the taker and makers...
  |      ^^^^ expected one of 8 possible tokens

error: aborting due to 4 previous errors

Couldn't compile the test.

failures:
    src/lib.rs - (line 77)

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.22s

error: doctest failed, to rerun pass `--doc`

**Comment by 007harshmahajan** [comment]
build and integration test works on macos
```
coinswap on git macos_build [$!>] is pkg v0.1.0 via rs v1.80.0
> cargo build --features=integration-test-macos-aarch
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.51s

coinswap on git macos_build [$!>] is pkg v0.1.0 via rs v1.80.0
> cargo build --features=build-macos-aarch
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.22s
 ```

**Review by mojoX911 — APPROVED** [review]
ACK bc8cbf7eaecf4fe8f53a93e868a932f258c876c5

**Review by Shourya742 — CHANGES_REQUESTED** [review]
ACK. Can you remove the merge commit and rebase to master. We are good to go with this.

**Comment by 007harshmahajan** [comment]
> ACK. Can you remove the merge commit and rebase to master. We are good to go with this.

All good with sync with master, no need of rebase, no changes

**Review by mojoX911 — APPROVED** [review]
ACK

**Review by Shourya742 — CHANGES_REQUESTED** [review]
nACK. I don't think this is working. On my MacBook, it's not running the integration test; it's compiling, but the integration test isn't running. You can check the CI logs to confirm—it's just skipping the integration test.

**Comment by 007harshmahajan** [comment]
https://app.warp.dev/block/g4mDqermk94xAvTH0GiaXi
please once have a looks

**Review by mojoX911 — COMMENTED** [review]
Add macos bitcoind binary in the bin folder.

feature-gate in test-frameowrk/mod.rs to use the macos bitcoind binary.

File: README.md [hunk 6]
```diff
@@ -46,12 +46,9 @@ CoinSwap is a rust implementation of a variant of atomic-swap protocol, using HT
 * [Developer's resources](/docs/dev-book.md)
 
 ## Build and Test
-Macos (M series) needs some feature for compilation 
-
 ```console
-$ cargo build --target=aarch64-apple-darwin  --features=build-macos-aarch
+$ cargo build
 ```
```
Comment (Shourya742) [inline README.md:51]: remove this section

**Review by Shourya742 — CHANGES_REQUESTED** [review]
ACK. One small nit.

File: README.md (hunk 6 above)
Comment (007harshmahajan) [reply README.md:51]: so like its just the basic build cmd, we can keep also ig ?

Comment (Shourya742) [reply README.md:51]: Thats the point just a basic build command.

**Review by Shourya742 — APPROVED** [review]
ACK.
